homeassistant/components/vallox/__init__.py

```diff
@@ -171,7 +173,15 @@ async def async_update_data() -> ValloxState:
     hass.data[DOMAIN] = {"client": client, "coordinator": coordinator, "name": name}
 
     async def _async_load_platform_delayed(*_: Any) -> None:
-        await coordinator.async_refresh()
+        # We need a successful update before loading the platforms, because platform init code
+        # derives the UUIDs from the data the coordinator fetches.
+        while True:
+            await coordinator.async_refresh()
+            if coordinator.last_update_success:
+                break
+
+            await asyncio.sleep(INITIAL_COORDINATOR_UPDATE_RETRY_INTERVAL_SECONDS)
```

Comment:
We should log **once** that it failed.  Ideally this gets canceled at the `EVENT_HOMEASSISTANT_STOP` event.

Reply:
Here is how config entries handle this case https://github.com/home-assistant/core/blob/dev/homeassistant/config_entries.py#L357

Reply:
Thanks! Pushed an update.